subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationClient.kt

```diff
@@ -231,14 +245,20 @@ internal class KubernetesReservationClient(
         var pods = kubernetesApi.getPods(deploymentName)
 
         @Suppress("EXPERIMENTAL_API_USAGE")
-        while (!podsChannel.isClosedForSend && pods.isNotEmpty()) {
-            pods.forEach { pod ->
-                podsChannel.send(pod)
-            }
+        while (pods.isNotEmpty()) {
```

Comment:
For example, we check from coroutine that `podsChannel.isClosed` and then send but another coroutine could concurrently close the channel
So we need to get a mutex and make those bunch of operations atomicaly